src/terminal/terminal.rs

```diff
@@ -441,12 +457,12 @@ where
             Viewport::Fullscreen => self.backend.clear_region(ClearType::All)?,
             Viewport::Inline(_) => {
                 self.backend
-                    .set_cursor(self.viewport_area.left(), self.viewport_area.top())?;
+                    .set_cursor_position(self.viewport_area.as_position())?;
                 self.backend.clear_region(ClearType::AfterCursor)?;
             }
             Viewport::Fixed(area) => {
-                for row in area.top()..area.bottom() {
-                    self.backend.set_cursor(0, row)?;
+                for y in area.top()..area.bottom() {
+                    self.backend.set_cursor_position(Position { x: 0, y })?;
```

Comment:
```suggestion
                for row in area.rows() {
                    self.backend.set_cursor_position(Position { x: 0, row.y })?;
```

Reply:
As only the y is needed the current approach seems simpler than to calculate a bunch of rects? #1299 reduces the performance overhead, but its still more complex than using a range of numbers.

Reply:
no problem with not doing this. To my eyes it looks cleaner and the performance difference is minimal (microseconds / nanoseconds). How many rows per second should the terminal be expected to be able to clear?